### PR TITLE
Implemented the restrictChatMember method

### PR DESCRIPTION
**Pull request by Emulator000**
Implemented a method that was missing

### CODE REVIEW

**Comment by Emulator000** [comment]
Any chance to get this merged?

**Comment by knsd** [comment]
Sorry for delay, I'll review this pr shortly.

File: raw/src/requests/restrict_chat_member.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/// Use this method to kick a user from a group or a supergroup.
+/// In the case of supergroups, the user will not be able to return to the group on
+/// their own using invite links, etc., unless unbanned first.
+/// The bot must be an administrator in the group for this to work.
+#[derive(Debug, Clone, PartialEq, PartialOrd, Serialize)]
+#[must_use = "requests do nothing unless sent"]
+pub struct RestrictChatMember {
+    chat_id: ChatRef,
+    user_id: UserId,
+    until_date: Option<i32>,
```
Comment (knsd) [inline raw/src/requests/restrict_chat_member.rs]: Why is this `i32`?

File: raw/src/requests/restrict_chat_member.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    type Type = JsonRequestType<Self>;
+    type Response = JsonTrueToUnitResponse;
+
+    fn serialize(&self) -> Result<HttpRequest, Error> {
+        Self::Type::serialize(RequestUrl::method("restrictChatMember"), self)
+    }
+}
+
+impl RestrictChatMember {
+    pub fn new<C, U>(chat: C, user: U, until_date: Option<i32>, can_send_messages: Option<bool>, can_send_media_messages: Option<bool>, can_send_other_messages: Option<bool>, can_add_web_page_previews: Option<bool>) -> Self where C: ToChatRef, U: ToUserId {
```
Comment (knsd) [inline raw/src/requests/restrict_chat_member.rs]: I think it's too many arguments for one method. How about `pub fn new<C, U>(chat: C, user: U)` and something like `pub fn can_send_messages(&mut self, value: bool) -> &mut Self`?

**Comment by Emulator000** [comment]
@knsd what do you think?

File: raw/src/requests/restrict_chat_member.rs (hunk 1 above)
Comment (Emulator000) [reply raw/src/requests/restrict_chat_member.rs]: It's the unix time date, maybe we have to use a better format?

File: raw/src/requests/restrict_chat_member.rs (hunk 2 above)
Comment (Emulator000) [reply raw/src/requests/restrict_chat_member.rs]: Fixed!

**Comment by knsd** [comment]
I've resolved conflicts and did some refactoring in #179. Unfortunately API was changed notably and it can't be merged right now.

**Pull request closed without merging**